src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -123,6 +124,29 @@ sol_str_slice_to_string(const struct sol_str_slice slice)
     return strndup(slice.data, slice.len);
 }
 
+static inline void
+sol_str_slice_remove_leading_whitespace(struct sol_str_slice *slice)
+{
+    while (isspace(*slice->data)) {
+        slice->data++;
+        slice->len--;
+    }
+}
+
+static inline void
+sol_str_slice_remove_trailing_whitespace(struct sol_str_slice *slice)
```

Comment:
Same here.

This is more consistent with many other usages of the string slice in APIs, where the structure is copied rather than passing a pointer to that structure. It's simple struct so it's pretty cheap.